Done

8Invader8 opened the pull request:
Done!

8Invader8 committed:
Done

Volodymyr95 commented on src/main/java/Horse.java:
```diff
@@ -1,27 +1,36 @@
+    private static final Logger logger = LogManager.getLogger(Horse.class);
     public Horse(String name, double speed, double distance) {
         if (isNull(name)) {
+            logger.info("Name is null");
```
use warn here

Volodymyr95 commented on src/main/java/Horse.java:
```diff
@@ -1,27 +1,36 @@
     private final double speed;
     private double distance;
-
+    private static final Logger logger = LogManager.getLogger(Horse.class);
```
make name in uppercase

Volodymyr95 commented on src/main/java/Horse.java:
```diff
@@ -1,27 +1,36 @@
+            logger.info("Name is null");
             throw new IllegalArgumentException("Name cannot be null.");
         } else if (name.isBlank()) {
+            logger.info("Name is blank");
```
use warn here

Volodymyr95 commented on src/main/java/Horse.java:
```diff
@@ -1,27 +1,36 @@
             throw new IllegalArgumentException("Name cannot be blank.");
         }
         if (speed < 0) {
+            logger.info("Speed is negative");
```
use warn here

Volodymyr95 commented on src/main/java/Horse.java:
```diff
@@ -1,27 +1,36 @@
             throw new IllegalArgumentException("Speed cannot be negative.");
         }
         if (distance < 0) {
+            logger.info("Distance is negative");
```
use warn here

Volodymyr95 commented on src/test/java/HippodromeTest.java:
```diff
@@ -0,0 +1,74 @@
+        hippodrome.move();
+
+        for (Horse hors : horses) {
+            Mockito.verify(hors, Mockito.times(1)).move();
```
use static import, verify() instead of Mockito.verify()

Volodymyr95 commented on src/test/java/HippodromeTest.java:
```diff
@@ -0,0 +1,74 @@
+    @Test
+    void getWinner_returnsCorrectWinner() {
+        Hippodrome hippodrome = new Hippodrome(List.of(
+                new Horse("horse5",1,5),
```
formatting

Volodymyr95 commented on src/test/java/HorseTest.java:
```diff
@@ -0,0 +1,114 @@
+    @Test
+    void getSpeed_ReturnsCorrectSpeed() {
+        String name = "TestName";
+        double speed = 1;
```
u are duplacate this variables on all tests, just create all in class level and reuse it

Volodymyr95 commented on src/test/java/MainTest.java:
```diff
@@ -0,0 +1,17 @@
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class MainTest {
```
not need this

8Invader8 committed:
Change:
-logget to LOGGER;
-logger info() to warn();
-some structure in Horse class;
-and minor things.
